src/WS2812FX/effects/PopcornEffect.cpp

```diff
@@ -163,9 +163,9 @@ void PopcornEffect::renderKernel(double position, KernelData& kernel, WS2812FX*
     uint8_t width = 1;
     uint8_t myWidth = (uint8_t)((double)(currentPos - prevPosition) / MM_PER_LED);
```

Comment:
Careful, according to the calculation, the result could be negative and myWidth being unsigned would have the wrong value. maybe the "-width" would fix this but I am not sure and it does not look "right". So maybe we just do:

```suggestion
if (currentPos > prevPosition) {
      width = (uint8_t)((double)(currentPos - prevPosition) / MM_PER_LED);
    }
    else
    {
      width = (uint8_t)((double)(prevPosition - currentPos) / MM_PER_LED);
    }
    if(!width)
    {
     width = 1;
    }
```